hoot-core/src/main/cpp/hoot/core/visitors/ReplaceTagVisitor.h

```diff
@@ -0,0 +1,102 @@
+/*
+ * This file is part of Hootenanny.
+ *
+ * Hootenanny is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * --------------------------------------------------------------------
+ *
+ * The following copyright notices are generated automatically. If you
+ * have a new notice to add, please use the format:
+ * " * @copyright Copyright ..."
+ * This will properly maintain the copyright information. DigitalGlobe
+ * copyrights will be updated automatically.
+ *
+ * @copyright Copyright (C) 2018 DigitalGlobe (http://www.digitalglobe.com/)
+ */
+#ifndef REPLACETAGVISITOR_H
+#define REPLACETAGVISITOR_H
+
+// hoot
+#include <hoot/core/visitors/ElementOsmMapVisitor.h>
+
+namespace hoot
+{
+
+/**
+ * @brief The ReplaceTagVisitor class replaces matching tags with the
+ * specified replacement tag.
+ *
+ * This was written for the building outline project: the input data had tag
+ * BUILDING=yes, but the output data needed to have tag building=yes
+ *
+ * Feel free to extend this class. Maybe add a flag for case-insensitive
+ * matching. Or support for wildcards.
+ */
+class ReplaceTagVisitor : public ElementOsmMapVisitor
```

Comment:
There is overlap between this class and SetTagVisitor used with _overwriteExistingTag = true, I think.  "Replace" is more readable in the class name for the behavior, though.  I guess later we could combine the two into one class, since SetTagVisitor has a few other useful features...not a big deal.

Reply:
Got it.  Didn't notice you were writing the key as well.

Reply:
I think I'm going to rename the other one to SetTagValueVisitor.